File: tax-manager-1.4.0-rc1/src/modules/core/TaxManager.Domain/Entities/Taxation.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace TaxManager.Domain.Entities;

[Index(nameof(Code), IsUnique = true)]
public class Taxation
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column(Order = 1)]
    public Guid Id { get; set; }
    
    [Required]
    [Column(TypeName = "varchar(40)")] //Initial value, length of properties may change.
    public string Code { get; set; } = string.Empty;
    
    [Required]
    public int CountryId { get; set; }
    
    [Required]
    [Column(TypeName = "varchar(255)")]
    public string Name { get; set; } = string.Empty;
    
    [ForeignKey("CountryId")]
    [DeleteBehavior(DeleteBehavior.Restrict)]
    public virtual Country Country { get; set; }
    
    public virtual List<Subdivision> Subdivisions { get; set; }= new List<Subdivision>();

    public virtual List<TaxationTaxRate> TaxationTaxRates { get; set; } = new List<TaxationTaxRate>();
    
    public virtual IEnumerable<LegalEnvironment> LegalEnvironments { get; set; }
    
}

File: tax-manager-1.4.0-rc1/src/hosts/TaxManager.Web/src/types/Taxation.ts
import { Subdivision } from './Subdivision';
import { Country } from './Country';
import { TaxationTaxRateDto } from "./TaxRate";

export interface Taxation {
    id?: number;
    code: string;
    countryId: number;
    name: string;
    subdivisions?: Subdivision[];
    country?: Country;
    taxationTaxRates: TaxationTaxRateDto[];
}

export interface TaxationResultDto {
    isValid: boolean;
    errors: string[];
    entity: Taxation;
}
